File: 백준/Bronze/23968. 알고리즘 수업 － 버블 정렬 1/알고리즘 수업 － 버블 정렬 1.py
import sys
input=lambda:sys.stdin.readline().rstrip()
n,k=map(int,input().split())
arr=[*map(int,input().split())]
cnt=0
for i in range(n):
    for j in range(n-1-i):
        if arr[j]>arr[j+1]:
            if cnt==k-1:
                print(arr[j+1],arr[j])
                exit(0)
            cnt+=1
            temp=arr[j]
            arr[j]=arr[j+1]
            arr[j+1]=temp
print(-1)

File: 백준/Bronze/23968. 알고리즘 수업 － 버블 정렬 1/알고리즘 수업 － 버블 정렬 1.c
#include <stdio.h>
#include <stdlib.h>
int main(void) {
    int n, k, cnt = 0, temp, *arr;
    scanf("%d %d", &n, &k);
    arr = (int*)malloc(sizeof(int) * n);
    for (int i = 0; i < n; i++)
        scanf("%d", &arr[i]);
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n - 1 - i; j++) {
            if (arr[j] > arr[j + 1]) {
                if (cnt == k - 1) {
                    printf("%d %d", arr[j + 1], arr[j]);
                    return 0;
                }
                cnt += 1;
                temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }
    printf("%d", -1);
    return 0;
}